src/lib/User/actions.ts

```diff
@@ -1,19 +1,19 @@
 "use server";
 
+import { EMAIL_ALREADY_EXISTS_MSG, signUpSchema } from "./validations";
 import { redirect } from "next/navigation";
 import { connectMongo } from "../db";
-import { createSession, setSessionCookie } from "../tools/session";
 import UserModel from "./model";
-import { EMAIL_ALREADY_EXISTS_MSG, signUpSchema } from "./validations";
 import { ZodIssue } from "zod";
+import bcrypt from "bcrypt";
 
 export async function signupUser(
   data: FormData,
 ): Promise<ZodIssue[] | unknown> {
-  const fullName = data.get("fullName");
-  const email = data.get("email");
-  const password = data.get("password");
-  const confirmPassword = data.get("confirmPassword");
+  const fullName = data.get("fullName") as string;
```

Comment:
We shouldn't be casting these

Reply:
Oh, i was getting an error prior to casting